google-cloud-datastore/src/test/java/com/google/cloud/datastore/it/ITDatastoreTest.java

```diff
@@ -137,6 +139,46 @@
   private static final Entity ENTITY3 = Entity.newBuilder(ENTITY1).setKey(KEY3).remove("str")
       .set("null", NULL_VALUE).set("partial1", PARTIAL_ENTITY2).set("partial2", ENTITY2).build();
 
+  private <T> Iterator<T> getStronglyConsistentResults (Query scQuery, Query query) {
+    QueryResults<T> scResults = DATASTORE.run(scQuery);
+    List<T> scResultsCopy = makeResultsCopy(scResults);
+    Set<T> scResultsSet = new HashSet<>(scResultsCopy);
+
+    infiniteloop:
+    while(true) {
```

Comment:
in case of any kind of unexpected problem this potentially can make the loop never succeed. Please consider putting maximum allowed attempts here, and throwing an exception if the limit is reached (so there will be no hanged tests).